src/DocBlock/Tags/Factory/MethodParameterFactory.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * This file is part of phpDocumentor.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ *
+ * @link      http://phpdoc.org
+ */
+
+namespace phpDocumentor\Reflection\DocBlock\Tags\Factory;
+
+use phpDocumentor\Reflection\DocBlock\Tags\Formatter;
+use function str_repeat;
+use function strlen;
+
+class MethodParameterFactory
```

Comment:
Please mark this class as final and `@internal` as we should not make this part of the backward compatibility promise. Extending this class should be blocked as it is an internal only class. Maybe this should even be part of the `Method` tag? Unless there is a reason to extract this because you want to share this class?

Reply:
All good

For me, extracting this logic from the base code is a good idea for the maintanability 
It could have been a trait but I did get inspired by the package way of working, could be wrong